src/main/java/com/khutircraftubackend/product/ProductRepository.java

```diff
@@ -12,65 +13,40 @@
-	 *
-	 * Використання to_tsvector та to_tsquery залежить від локалі бази даних (uk_UA.UTF-8).
-	 */
-	@Query(value = """
-			    SELECT p.*
-			    FROM products p
-			    JOIN categories c ON p.category_id = c.id
-			    WHERE p.available = true AND (
-			        to_tsvector('simple', c.name) @@ to_tsquery('simple', :keyword) OR
-			        to_tsvector('simple', p.name) @@ to_tsquery('simple', :keyword)
-			        )
-			    ORDER BY
-			        CASE
-			            WHEN to_tsvector('simple', c.name) @@ to_tsquery('simple', :keyword) THEN 1
-			            WHEN to_tsvector('simple', p.name) @@ to_tsquery('simple', :keyword) THEN 2
-			        END
-			""", nativeQuery = true)
-	List<ProductEntity> searchWithPriority(@Param("keyword") String keyword);
-	
-	/**
-	 * Здійснює пошук підказок для автозаповнення.
-	 * Пошук виконується з ігноруванням регістру серед назв продуктів, категорій та описів.
-	 * Результати унікальні, відсортовані за назвою, з обмеженням до 10 записів.
-	 *
-	 * @param query Ключове слово для пошуку. Використовується для пошуку збігів, що починаються з введеного слова.
-	 * @return Список унікальних рядків, що відповідають критеріям пошуку, відсортованих за алфавітом.
-	 *
-	 * Використання ILIKE залежить від локалі бази даних (uk_UA.UTF-8).
-	 */
-	@Query(value = """
-			    SELECT DISTINCT name
-			             FROM (
-			                 SELECT p.name AS name
-			                 FROM products p
-			                 JOIN categories c ON p.category_id = c.id
-			                 WHERE c.name ILIKE (CONCAT('%', :query, '%')) OR
-			                       p.name ILIKE (CONCAT('%', :query, '%'))
-			             ) suggestions /*тимчасова таблиця*/
-			             ORDER BY name
-			             LIMIT 10;
-			""", nativeQuery = true)
-	List<String> findSuggestions(@Param("query") String query);
+    
+    List<ProductEntity> findAllBy(Pageable pageable);
+    
+    Optional<ProductEntity> findProductById(Long id);
+    
+    void deleteBySeller(SellerEntity seller);
+    
+    List<ProductEntity> findAllBySeller(SellerEntity currentSeller);
+    
+    @Query(value = """
+  with recursive catalogue as (
+  -- root categories:
+  select c.id, concat(c.name, ',', coalesce(c.keywords, ''))::text as "text"
+  from categories c where parent_id is null
+  -- sub categories:
+  union all
+  select c.id, concat(ctg.text, ',', c.name, ',', coalesce(c.keywords, ''))::text as "text"
+  from catalogue ctg
+  inner join categories c on c.parent_id = ctg.id), suggestions as (
+  select distinct p.id, p.name,
+    to_tsvector('simple', concat(p.name, ' ', coalesce(c."text", ''))) as "tsvector",
+    to_tsquery('simple', concat(replace(:query, ' ', ':* & '), ':*')) as "tsquery"
+  from products p
+  inner join catalogue c on c.id = p.category_id
+  where p.available = true
+  )
+  select s.id, s.name,
+  ts_rank_cd("tsvector", "tsquery") as rank
+    from suggestions s
+    where "tsvector" @@ "tsquery"
+    order by rank desc
+    limit 50;
```

Comment:
minor suggestion to reformat code for better readability: 
```suggestion
  with recursive catalogue as (
    -- root categories:
    select
      c.id, concat(c.name, ',', coalesce(c.keywords, ''))::text as "text"
    from categories c 
    where parent_id is null
    -- 
    union all
    -- sub categories:
    select
      c.id, concat(ctg.text, ',', c.name, ',', coalesce(c.keywords, ''))::text as "text"
    from catalogue ctg
    inner join categories c on c.parent_id = ctg.id
  ), suggestions as (
    select
      distinct p.id, p.name,
      to_tsvector('simple', concat(p.name, ' ', coalesce(c."text", ''))) as "tsvector",
      to_tsquery('simple', concat(replace(:query, ' ', ':* & '), ':*')) as "tsquery"
    from products p
    inner join catalogue c on c.id = p.category_id
    where p.available = true
  )
  select
    s.id, s.name,
    ts_rank_cd("tsvector", "tsquery") as rank
  from suggestions s
  where "tsvector" @@ "tsquery"
  order by rank desc
  limit 50;
```

Reply:
Tab settings are different for everyone, so there may be a difference.

Reply:
this is not about tabs :) 
compare this line 
`inner join categories c on c.parent_id = ctg.id), suggestions as (` in your code 
with my suggestion :)

Reply:
these are, in general, subjective views.